internal/executor/instance/persistentworker/isolation/tart/vm.go

```diff
@@ -28,6 +35,20 @@ func NewVMClonedFrom(ctx context.Context, from string, cpu uint32, memory uint32
 		errChan:      make(chan error, 1),
 	}
 
+	pullLogger := logger.Scoped("pull virtual machine")
+	if !lazyPull {
+		pullLogger.Infof("Pulling virtual machine %s...", from)
+
+		if _, _, err := CmdWithLogger(ctx, pullLogger, "pull", from); err != nil {
+			pullLogger.FinishWithType(echelon.FinishTypeSucceeded)
+			return nil, err
+		}
+
+		pullLogger.FinishWithType(echelon.FinishTypeSucceeded)
+	} else {
+		pullLogger.FinishWithType(echelon.FinishTypeSkipped)
```

Comment:
Do we need to print anything? What do you think of not even creating the scope?

Reply:
I think it provides better transparency into how the process could be, so that users might discover a new CLI option simply by looking at the skipped steps.